.github/workflows/check-clippy-all-features.yml

```diff
@@ -0,0 +1,58 @@
+name: Check Clippy all-features
+
+on:
+  pull_request:
+
+jobs:
+  get-head-commit-msg:
+    runs-on: ubuntu-latest
+    outputs:
+      head-commit-msg: ${{ steps.step-head-commit.outputs.headCommitMsg }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get Head Commit Message
+        id: step-head-commit
+        run: echo "headCommitMsg=$(git show -s --format=%s)" >> "$GITHUB_OUTPUT"
+
+  check-clippy-all-features:
```

Comment:
We can do a matrix strategy as in https://github.com/kilt-jam/jam-client/blob/master/.github/workflows/clippy.yaml.

Reply:
nice, applied it in tests as well